packages/std/src/math/uint256.rs

```diff
@@ -0,0 +1,560 @@
+                rhs
+            );
+        }
+
+        Self(self.0.shr(rhs))
+    }
+}
+
+impl<'a> ops::Shr<&'a u32> for Uint256 {
+    type Output = Self;
+
+    fn shr(self, rhs: &'a u32) -> Self::Output {
+        if *rhs > 256 {
+            panic!(
+                "right shift error: {} is larger than the number of bits in Uint256",
+                rhs
+            );
+        }
+
+        Self(self.0.shr(*rhs))
+    }
+}
+
+impl ops::AddAssign<Uint256> for Uint256 {
+    fn add_assign(&mut self, rhs: Uint256) {
+        self.0 = self.0.checked_add(rhs.0).unwrap();
+    }
+}
+
+impl<'a> ops::AddAssign<&'a Uint256> for Uint256 {
+    fn add_assign(&mut self, rhs: &'a Uint256) {
+        self.0 = self.0.checked_add(rhs.0).unwrap();
+    }
+}
+
+impl ops::SubAssign<Uint256> for Uint256 {
+    fn sub_assign(&mut self, rhs: Uint256) {
+        self.0 = self.0.checked_sub(rhs.0).unwrap();
+    }
+}
+
+impl<'a> ops::SubAssign<&'a Uint256> for Uint256 {
+    fn sub_assign(&mut self, rhs: &'a Uint256) {
+        self.0 = self.0.checked_sub(rhs.0).unwrap();
+    }
+}
+
+impl ops::DivAssign<Uint256> for Uint256 {
+    fn div_assign(&mut self, rhs: Self) {
+        self.0 = self.0.checked_div(rhs.0).unwrap();
+    }
+}
+
+impl<'a> ops::DivAssign<&'a Uint256> for Uint256 {
+    fn div_assign(&mut self, rhs: &'a Uint256) {
+        self.0 = self.0.checked_div(rhs.0).unwrap();
+    }
+}
+
+impl ops::ShrAssign<u32> for Uint256 {
+    fn shr_assign(&mut self, rhs: u32) {
+        if rhs > 256 {
+            panic!(
+                "right shift error: {} is larger than the number of bits in Uint256",
+                rhs
+            );
+        }
+
+        self.0 = self.0.shr(rhs);
+    }
+}
+
+impl<'a> ops::ShrAssign<&'a u32> for Uint256 {
+    fn shr_assign(&mut self, rhs: &'a u32) {
+        if *rhs > 256 {
+            panic!(
+                "right shift error: {} is larger than the number of bits in Uint256",
+                rhs
+            );
+        }
+
+        self.0 = self.0.shr(*rhs);
+    }
+}
+
+impl Serialize for Uint256 {
+    /// Serializes as an integer string using base 10
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: ser::Serializer,
+    {
+        serializer.serialize_str(&self.to_string())
+    }
+}
+
+impl<'de> Deserialize<'de> for Uint256 {
+    /// Deserialized from an integer string using base 10
+    fn deserialize<D>(deserializer: D) -> Result<Uint256, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_str(Uint256Visitor)
+    }
+}
+
+struct Uint256Visitor;
+
+impl<'de> de::Visitor<'de> for Uint256Visitor {
+    type Value = Uint256;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("string-encoded integer")
+    }
+
+    fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+    where
+        E: de::Error,
+    {
+        match U256::from_dec_str(v) {
```

Comment:
Can we use our `impl TryFrom<&str> for Uint256` from above to avoid having different string parsers?

Reply:
Just for the record, I initially changed this from using `parse` because the inner `uint::U256` type seems to expect a hexadecimal representation by default. I hadn't noticed the `TryFrom<&str>` thing. Fixed now.